pkgs/development/libraries/openmpi/default.nix

```diff
@@ -49,6 +56,16 @@ in stdenv.mkDerivation rec {
     rm -f $out/lib/*.la
    '';
 
+  postFixup = ''
+    # default compilers should be indentical to the
+    # compilers at build time
+
+    sed -i 's:compiler=.*:compiler=${stdenv.cc}/bin/cc:' $out/share/openmpi/mpicc-wrapper-data.txt
```

Comment:
Is this correct in the case of cross-compilation?

Reply:
Cross compilation is still a mystery to me but this is probably only works if `hostPlatform == buildPlatform`. How would I get the compilers of the target platform? cc @Mic92

Reply:
@markuskowa I think this would be `${targetPackages.stdenv.cc}`.

Reply:
But actually there was this one recently introduced: https://github.com/NixOS/nixpkgs/pull/57611
So instead this would be:

```
"${pkgsTargetTarget.stdenv.cc}/bin/${pkgsTargetTarget.stdenv.cc.targetPrefix}cc"
```

Reply:
@Ericson2314 could maybe also provide the right expression for `gfortran`.

Reply:
Thanks for the insight! Just extrapolating from the expression for gcc:
```
${pkgsTargetTarget.gfortran}/bin/${pkgsTargetTarget.gfortran.targetPrefix}gfortran
```
Does that make sense?